assets/js/type-defs/registered-payment-method-props.js

```diff
@@ -0,0 +1,206 @@
+ * @property {Function} completed
+ * @property {Function} error
+ * @property {Function} failed
+ * @property {Function} success
+ */
+
+/**
+ * @typedef {Object} PaymentMethodShipping
+ *
+ * @property {boolean}              required      If shipping is required this
+ *                                                will be true.
+ * @property {CartShippingOption[]} options       Available shipping options.
+ * @property {Function}             selectOptions Used to set the selected
+ *                                                shipping options.
+ * @property {Function}             updateAddress Used to update the shipping
+ *                                                address.
+ * @property {Function}             setStatus     Used to set the current
+ *                                                shipping options status.
+ * @property {Function}             selectStatus  Returns helpers for
+ *                                                determining the current
+ *                                                shipping options status.
+ * @property {string}               status        What the current shipping
+ *                                                options status is.
+ */
+
+/**
+ * @typedef {Object} PaymentMethodEvents
+ *
+ * @property {boolean}  isCalculating      If true, means the cart/checkout is
+ *                                         currently calculating totals.
+ * @property {boolean}  isCheckoutComplete If true, means the checkout process
+ *                                         is complete (useful if the payment
+ *                                         method has something to do after
+ *                                         checkout is complete before setting
+ *                                         payment status to complete. Redirect
+ *                                         doesn't happen until both checkout
+ *                                         and payment status is complete)
+ * @property {PaymentStatusActions} dispatchStatus     Used to dispatch a payment event.
+ * @property {Function} selectStatus       Returns helpers for determining
+ *                                         current payment event status.
+ * @property {string}   status             What the current payment event status
+ *                                         is.
+ */
+
+/**
+ * @typedef CheckoutStatusProps
+ *
+ * @property {boolean} isCalculating If true then totals are being calculated in
+ *                                   the checkout.
+ * @property {boolean} isComplete    If true then the checkout has completed
+ *                                   it's processing.
+ * @property {boolean} isIdle        If true then the checkout is idle (no
+ *                                   activity happening).
+ * @property {boolean} isProcessing  If true then checkout is processing
+ *                                   (finalizing) the order with the server.
+ */
+
+/**
+ * @typedef PaymentStatusProps
```

Comment:
Would be great to have a one-liner for each of these types clearly defining the purpose. Sometimes that's more useful than the individual fields. I realise this is not an easy task, we can iterate on this as follow up :)

Reply:
Sure can do in a follow-up where more explanation is needed.